esphome/components/udp/__init__.py

```diff
@@ -0,0 +1,159 @@
+import hashlib
+
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components.api import CONF_ENCRYPTION
+from esphome.components.binary_sensor import BinarySensor
+from esphome.components.sensor import Sensor
+from esphome.const import (
+    CONF_PORT,
+    CONF_SENSORS,
+    CONF_ID,
+    CONF_BINARY_SENSORS,
+    CONF_NAME,
+    CONF_KEY,
+    PLATFORM_ESP32,
+    PLATFORM_BK72XX,
+    PLATFORM_RTL87XX,
+    PLATFORM_HOST,
+    CONF_INTERNAL,
+)
+from esphome.cpp_generator import MockObjClass
+
+CODEOWNERS = ["@clydebarrow"]
+DEPENDENCIES = ["network"]
+AUTO_LOAD = ["socket", "sensor", "binary_sensor"]
```

Comment:
Please use the defines to not autoload platform_components
```suggestion
AUTO_LOAD = ["socket"]
```